src/pages/home/home.ts

```diff
@@ -15,9 +15,11 @@ export class HomePage {
   constructor(private navCtrl: NavController, push: PushService, plt: Platform, public auth: Auth) {
     // We need to wait for the platform to initialize the plugins
     plt.ready().then(() => {
-      push.initPush();
-      push.setCallback((n) => this.addNotification(n));
-      push.register();
+      if (!PushService.registered) {
```

Comment:
On IOS not registration happens twice. When initially we register we still do not get user permission. Then on user permission we are getting token again. This is some quirk with the plugin implementation, but I'm worried that caching registrations here can lead to errors. IMHO it's best to register every time. Any possible side effects?

Reply:
There should be no problem with registering multiple times, as long as the deviceId doesn't change.